test/e2e_node/criproxy/image.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package criproxy
+
+import (
+	"context"
+
+	internalapi "k8s.io/cri-api/pkg/apis"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+type imageServiceProxy struct {
+	errorInjector func(string) error
```

Comment:
When you pass the name of the CRI API to `errorInjector`, does that mean we can only inject errors into specific CRI APIs? Should we also use `errorInjector` when injecting a delay? Also, passing the context along might make it easier to inject a timeout of CRI call.

Reply:
For https://github.com/kubernetes/kubernetes/pull/121604, we need to add a delay for image pulling to 10s or more.

Reply:
> When you pass the name of the CRI API to errorInjector, does that mean we can only inject errors into specific CRI APIs?

no, we can dynamically inject errors through SetErrorInjectors..

Reply:
> Should we also use errorInjector when injecting a delay?

yes, my initial idea was to use errorInjector for both. I decided to separate them because it improves readability and flexibility.

Reply:
> Also, passing the context along might make it easier to inject a timeout of CRI call.

Injecting delays through `context` is not a common or recommended approach; `context` is better suited for timeouts. Using `time.Sleep` explicitly in `delayInjector` is simpler and more straightforward.